corehq/apps/hqwebapp/views.py

```diff
@@ -395,6 +396,24 @@ def debug_notify(request):
             "If you want to achieve a 500-style email-out but don't want the user to see a 500, use notify_exception(request[, message])")
     return HttpResponse("Email should have been sent")
 
+
+@require_POST
+def jserror(request):
+    stack = request.POST.get('stack', None)
+
+    ExponentialBackoff.increment(stack)
+    if not ExponentialBackoff.should_backoff(stack):
+        notify_js_exception(
+            request,
+            message=request.POST.get('message', None),
+            stack=stack,
+            line=request.POST.get('line', None),
+            filename=request.POST.get('filename', None),
+        )
```

Comment:
I think this is fine, other option is to funnel to sentry, but I don't see a strong concrete difference one way or the other.

Other thing I'd mention is that if you key off the whole stack there's a high potential that you're going to get "the same error" under a number of different stack traces, effectively increasing the number of emails by a (high?) constant factor. 
(There may be many ways to get to offending function F, and they'll each have a different stack trace.) I'm not sure this is going to be a problem but just wanted to raise that if you get way more volume on each single thread than you thought given exponential backoff, keying off just the last N lines of the stacktrace could help.

The other thing I'd add that's very useful to the email is the number of times an error has been triggered. Soft assert does this and that number is a super useful heuristic for how much something is "blowing up"